src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -923,11 +927,29 @@ static void ExpandStars(double targetSize, double currentSize, Definition[] defs
 					// targetStarSize, that means we have enough room to expand all of our star rows/columns
 					// to their full size.
 
-					foreach (var definition in defs)
+					// Use density-aware distribution for pixel-perfect allocation when density != 1.0
+					// For density = 1.0, preserve original behavior to maintain fractional values
+					if (Math.Abs(density - 1.0) > 0.0001)
 					{
-						if (definition.IsStar)
+						var starDefinitions = defs.Where(d => d.IsStar).ToArray();
+						var portions = starDefinitions.Select(d => targetStarSize * d.GridLength.Value).ToArray();
+						var totalPixels = portions.Sum() * density;
+						var pixelAllocations = DensityValue.DistributePixels(totalPixels, density, portions);
+						
+						for (int i = 0; i < starDefinitions.Length; i++)
+						{
+							starDefinitions[i].Size = DensityValue.FromPixels(pixelAllocations[i], density);
+						}
```

Comment:
I want this code to be the code that runs regardless of what the density value is. 

We shouldn't process the calculation of the sizes differently if the density equals 1 vs if it's greater than 1.